cfg/boost.cfg

```diff
@@ -14,4 +14,10 @@
       <not-null/>
     </arg>
   </function>
+  <!-- https://www.boost.org/doc/libs/master/libs/config/doc/html/boost_config/boost_macro_reference.html -->
+  <define name="BOOST_FORCEINLINE" value="inline"/>
+  <define name="BOOST_NOINLINE" value=""/>
+  <define name="BOOST_NORETURN" value=""/>
```

Comment:
should this value be `__attribute__((noreturn))` perhaps?

Reply:
Good point, yes it uses this value or some similar annotation.
My Boost installation defines it this way:
```
// BOOST_NORETURN ---------------------------------------------//
// Macro to use before a function declaration/definition to designate
// the function as not returning normally (i.e. with a return statement
// or by leaving the function scope, if the function return type is void).
#if !defined(BOOST_NORETURN)
#  if defined(_MSC_VER)
#    define BOOST_NORETURN __declspec(noreturn)
#  elif defined(__GNUC__)
#    define BOOST_NORETURN __attribute__ ((__noreturn__))
#  elif defined(__has_attribute) && defined(__SUNPRO_CC) && (__SUNPRO_CC > 0x5130)
#    if __has_attribute(noreturn)
#      define BOOST_NORETURN [[noreturn]]
#    endif
#  elif defined(__has_cpp_attribute) 
#    if __has_cpp_attribute(noreturn)
#      define BOOST_NORETURN [[noreturn]]
#    endif
#  endif
#endif

#if !defined(BOOST_NORETURN)
#  define BOOST_NO_NORETURN
#  define BOOST_NORETURN
#endif
```

Is it ok for Cppcheck to always use `__attribute__ ((__noreturn__))` as value?